### PR TITLE
Replace reflect.DeepEqual by equality.Semantic.DeepEqual

### PR DESCRIPTION
**Pull request by lekaf974**
Fix issue #3488

### CODE REVIEW

File: pkg/scheduler/uthelper/helper.go [hunk 1]
```diff
@@ -18,11 +18,11 @@ package uthelper
 
 import (
 	"fmt"
-	"reflect"
 	"time"
 
 	v1 "k8s.io/api/core/v1"
 	schedulingv1 "k8s.io/api/scheduling/v1"
+	quality "k8s.io/apimachinery/pkg/api/equality"
```
Comment (Monokaix) [inline pkg/scheduler/uthelper/helper.go:25]: just keep `equality` is ok: )

File: pkg/cli/queue/delete_test.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func TestDeleteQueue(t *testing.T) {
 		deleteQueueFlags.Name = testCase.QueueName
 
 		err := DeleteQueue()
-		if false == quality.Semantic.DeepEqual(err, testCase.ExpectValue) {
+		if false == reflect.DeepEqual(err, testCase.ExpectValue) {
```
Comment (Monokaix) [inline pkg/cli/queue/delete_test.go:74]: so why this failed?

**Comment by hwdef** [comment]
please squash your commit

File: pkg/cli/queue/delete_test.go (hunk 2 above)
Comment (lekaf974) [reply pkg/cli/queue/delete_test.go:74]: will revalidate but failing tests arrived  with `nil`

**Comment by lekaf974** [comment]
> please squash your commit

will do

File: pkg/cli/queue/delete_test.go (hunk 2 above)
Comment (lekaf974) [reply pkg/cli/queue/delete_test.go:74]: @Monokaix 

got a panic on second test 

debugging step by step **err** and **testCase.ExpectValue** are both type `*errors.ErrorString` with same value `queue name must be specified`


see full stacktrace

```bash
=== RUN   TestDeleteQueue
--- FAIL: TestDeleteQueue (85.95s)
=== RUN   TestDeleteQueue
--- FAIL: TestDeleteQueue (100.24s)
panic: an unexported field was encountered, nested like this:  [recovered]
	panic: an unexported field was encountered, nested like this: string [recovered]
	panic: an unexported field was encountered, nested like this: errors.errorString -> string [recovered]
	panic: an unexported field was encountered, nested like this: *errors.errorString -> errors.errorString -> string [recovered]
	panic: an unexported field was encountered, nested like this: *errors.errorString -> errors.errorString -> string

goroutine 14 [running]:
testing.tRunner.func1.2({0x1e4d860, 0xc000012168})
	/home/matthieu/go/sdk/go1.21.10/src/testing/testing.go:1545 +0x4b1
testing.tRunner.func1()
	/home/matthieu/go/sdk/go1.21.10/src/testing/testing.go:1548 +0x6a9
panic({0x1e4d860?, 0xc000012168?})
	/home/matthieu/go/sdk/go1.21.10/src/runtime/panic.go:920 +0x290
k8s.io/apimachinery/third_party/forked/golang/reflect.makeUsefulPanic({0x1e21840, 0xc00028c070, 0x16})
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96 +0x2df
panic({0x1e4d860?, 0xc000012150?})
	/home/matthieu/go/sdk/go1.21.10/src/runtime/panic.go:920 +0x290
k8s.io/apimachinery/third_party/forked/golang/reflect.makeUsefulPanic({0x1e60f80, 0xc00028c070, 0x199})
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96 +0x2df
panic({0x1e4d860?, 0xc000012138?})
	/home/matthieu/go/sdk/go1.21.10/src/runtime/panic.go:920 +0x290
k8s.io/apimachinery/third_party/forked/golang/reflect.makeUsefulPanic({0x1db7920, 0xc00028c070, 0x1b8})
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96 +0x2df
panic({0x1e4d860?, 0xc000012120?})
	/home/matthieu/go/sdk/go1.21.10/src/runtime/panic.go:920 +0x290
k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual(0xc000395020, {0x1db7920, 0xc00028c070, 0x1b8}, {0x1db7920, 0xc00011ea70, 0x1b8}, 0xc0001bba58, 0x1, 0x2)
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:257 +0x1486
k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual(0xc000395020, {0x1e60f80, 0xc00028c070, 0x199}, {0x1e60f80, 0xc00011ea70, 0x199}, 0xc0001bba58, 0x1, 0x1)
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:206 +0x1565
k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepValueEqual(0xc000395020, {0x1e21840, 0xc00028c070, 0x16}, {0x1e21840, 0xc00011ea70, 0x16}, 0xc0001bba58, 0x1, 0x0)
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:203 +0xedd
k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.deepEqual(0xc000395020, {0x1e21840, 0xc00028c070}, {0x1e21840, 0xc00011ea70}, 0x1)
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:289 +0x2b3
k8s.io/apimachinery/third_party/forked/golang/reflect.Equalities.DeepEqual(0xc000395020, {0x1e21840, 0xc00028c070}, {0x1e21840, 0xc00011ea70})
	/home/matthieu/Repos/github/volcano-sh/volcano/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:273 +0x50
volcano.sh/volcano/pkg/cli/queue.TestDeleteQueue(0xc000582820)
	/home/matthieu/Repos/github/volcano-sh/volcano/pkg/cli/queue/delete_test.go:74 +0x53b
testing.tRunner(0xc000582820, 0x20e6f38)
	/home/matthieu/go/sdk/go1.21.10/src/testing/testing.go:1595 +0x1f3
created by testing.(*T).Run in goroutine 1
	/home/matthieu/go/sdk/go1.21.10/src/testing/testing.go:1648 +0x7e5
```

`err` and `testCase.ExpectValue` are both `*errors.ErrorString` with same value `queue name must be specified`

**Comment by lekaf974** [comment]
> please squash your commit

@hwdef  done

File: pkg/scheduler/uthelper/helper.go (hunk 1 above)
Comment (lekaf974) [reply pkg/scheduler/uthelper/helper.go]: done

File: cmd/controller-manager/app/options/options_test.go [hunk 3]
```diff
@@ -17,11 +17,11 @@ limitations under the License.
 package options
 
 import (
-	"reflect"
 	"testing"
 	"time"
 
 	"github.com/spf13/pflag"
+	equality "k8s.io/apimachinery/pkg/api/equality"
```
Comment (Monokaix) [inline cmd/controller-manager/app/options/options_test.go:24]: 1. pkg name is already `equality` so there is no need to add a alias: )

Comment (lekaf974) [reply cmd/controller-manager/app/options/options_test.go:24]: though I removed this everywhere, will clean this soon

Comment (lekaf974) [reply cmd/controller-manager/app/options/options_test.go]: updated

**Comment by william-wang** [comment]
@Monokaix The `E2E Parallel Jobs` and `Scheduling actions` ci cases failed to start. please have a look.

**Comment by Monokaix** [comment]
> @Monokaix The `E2E Parallel Jobs` and `Scheduling actions` ci cases failed to start. please have a look.

Seems it's canceled, we can we re trigger it.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**